### PR TITLE
Major POSt in React

### PR DESCRIPTION
**Pull request by christinem**
This PR adds in the React code for the Major POSt. I created a general 'Post' component which all other POSt components can extend off of, since there would have been a lot of repeated code otherwise - I passed in everything that differs between the POSts as props. @david-yz-liu let me know what you think about this method.

### CODE REVIEW

File: public/js/post/post_components.js [hunk 1]
```diff
@@ -89,40 +76,120 @@ var SpecialistPost = React.createClass({
...
-                    categoryName='Any 300+ level CSC course, BCB410H, BCB420H, BCB430Y, ECE385H, ECE489H (1.5 FCEs)' />
-                <MultipleCourseCode courseID="spec_extra" textBoxNumber={4} courses={courseCategoryArrays[2]} textboxesDisabled={false} changeCourseCredit={this.changeCreditCount}
-                    categoryName='Any of the following: 300+ level CSC course; MAT: 235/237/257, any 300+ 
-                                  except for 329, 390, & 391; STA: 248, 261, any 300+; ECE: 385H/489H; 
-                                  BCB: 410H/420H/430Y (2.0 FCEs)' />  
+            <div id={'post_' + this.props.postType}>
+                <CourseCategory yearName='First Year' courses={this.props.firstYearCourses} />
+                <CourseCategory yearName='Second Year' courses={this.props.secondYearCourses} />
+                <CourseCategory yearName='Later Years' courses={this.props.laterYearCourses} />
+                <MultipleCourseCode courseID='spec_400' textBoxNumber={this.props.textBoxNumbers[0]} courses={courseCategoryArrays[0]} textboxesDisabled={true}
```
Comment (david-yz-liu) [inline public/js/post/post_components.js]: The `courseID`s need to be changed as well.

File: public/js/post/post_components.js [hunk 2]
```diff
@@ -89,40 +76,120 @@ var SpecialistPost = React.createClass({
...
+        var notes = ['No more than 1.0 FCE from CSC490H1, CSC491H1, CSC494H1, CSC495H1, BCB430Y1 may be used \
+                     to fulfill program requirements'];
+
+        var firstYearCourses = [['csc108'], ['csc148'], ['csc165', 'csc240'], ['mat135', 'mat136', 'mat137', 'mat157']];
+        var secondYearCourses = [['csc207'], ['csc209'], ['csc236', 'csc240'], ['csc258'], ['csc263', 'csc265'], ['mat221', 'mat223', 'mat240'], 
+                                ['sta247', 'sta255', 'sta257']];
+        var laterYearCourses = [['csc369'], ['csc373']];
+
+        return (
+            <div id ='specialist_window'>
```
Comment (david-yz-liu) [inline public/js/post/post_components.js]: Probably put this outer `div` inside the Post component.

**Comment by david-yz-liu** [comment]
@christinem this is a great approach!

**Comment by christinem** [comment]
@david-yz-liu awesome, thanks! Also made the fixes you mentioned.

**Pull request merged**